internal/resource/config/serversettings/generalsettings/server_settings_general_settings_data_source.go

```diff
@@ -0,0 +1,124 @@
+	SpConnectionTransactionLoggingOverride  types.String `tfsdk:"sp_connection_transaction_logging_override"`
+	DatastoreValidationIntervalSecs         types.Int64  `tfsdk:"datastore_validation_interval_secs"`
+	RequestHeaderForCorrelationId           types.String `tfsdk:"request_header_for_correlation_id"`
+}
+
+// GetSchema defines the schema for the datasource.
+func (r *serverSettingsGeneralSettingsDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	schema := schema.Schema{
+		Description: "Describes ServerSettingsGeneralSettings.",
+		Attributes: map[string]schema.Attribute{
+			"datastore_validation_interval_secs": schema.Int64Attribute{
+				Description: "How long (in seconds) the result of testing a datastore connection is cached.",
+				Computed:    true,
+				Optional:    false,
+			},
+			"disable_automatic_connection_validation": schema.BoolAttribute{
+				Description: "Boolean that disables automatic connection validation when set to true.",
+				Computed:    true,
+				Optional:    false,
+			},
+			"idp_connection_transaction_logging_override": schema.StringAttribute{
+				Description: "Describes the level of transaction logging for all identity provider connections. The default is DONT_OVERRIDE, in which case the logging level will be determined by each individual IdP connection [ DONT_OVERRIDE, NONE, FULL, STANDARD, ENHANCED ]",
+				Computed:    true,
+				Optional:    false,
+			},
+			"request_header_for_correlation_id": schema.StringAttribute{
+				Description: "HTTP request header for retrieving correlation ID.",
+				Computed:    true,
+				Optional:    false,
+			},
+			"sp_connection_transaction_logging_override": schema.StringAttribute{
+				Description: "Determines the level of transaction logging for all service provider connections. The default is DONT_OVERRIDE, in which case the logging level will be determined by each individual SP connection [ DONT_OVERRIDE, NONE, FULL, STANDARD, ENHANCED ]",
+				Computed:    true,
+				Optional:    false,
+			},
+		},
+	}
+
+	id.ToSchema(&schema)
+	resp.Schema = schema
+}
+
+// Metadata returns the resource type name.
+func (r *serverSettingsGeneralSettingsDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_server_settings_general_settings"
+}
+
+func (r *serverSettingsGeneralSettingsDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, _ *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	providerCfg := req.ProviderData.(internaltypes.ResourceConfiguration)
+	r.providerConfig = providerCfg.ProviderConfig
+	r.apiClient = providerCfg.ApiClient
+
+}
+
+func readServerSettingsGeneralSettingsDataSource(ctx context.Context, r *client.GeneralSettings, state *serverSettingsGeneralSettingsDataSourceModel) {
+	state.Id = types.StringValue("serverSettingsGeneralSettingsId")
```

Comment:
I'd use the typical terraform underscore case for this. `server_settings_general_settings_id`

Reply:
done